airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -855,7 +855,7 @@ def parse_response(
 
         # Not all (partial) responses include the request date, so adding it manually here
         for record in document_records:
-            if stream_slice.get("dataEndTime"):
+            if stream_slice and stream_slice.get("dataEndTime"):
```

Comment:
Please add a unit test to validate your change. Probably adding an example of what is causing the issue.